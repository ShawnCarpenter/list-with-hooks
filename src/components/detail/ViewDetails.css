.ViewDetails {
  
  img {
    border-radius: .3rem;
  }
}
