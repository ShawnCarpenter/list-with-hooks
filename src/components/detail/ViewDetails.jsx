import React from 'react';
import PropTypes from 'prop-types';
import styles from './ViewDetails.css';

const ViewDetails = ({ name, image, occupation, description }) => {
  
  return (
    <div className={styles.ViewDetails}>
      <h1>{name}</h1>
      <img src={image} alt={name} />
      {occupation && <div>Occupation: {occupation}</div> }
      <div>{description}</div>
    </div>
  );
};

ViewDetails.propTypes = {
  match:PropTypes.object,
  name:PropTypes.string.isRequired,
  image:PropTypes.string.isRequired,
  occupation:PropTypes.string,
  description:PropTypes.string.isRequired
};

export default ViewDetails;
