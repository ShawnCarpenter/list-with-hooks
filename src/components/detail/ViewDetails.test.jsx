import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ViewDetails from './ViewDetails';

describe('ViewDetails component', () => {
  afterEach(() => cleanup());
  it('renders ViewDetails', () => {
    const { asFragment } = render(<ViewDetails
      name="Bob" 
      image="http://img.com/pic.png"
      occupation="builder"
      description="Weird stuff happened"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
