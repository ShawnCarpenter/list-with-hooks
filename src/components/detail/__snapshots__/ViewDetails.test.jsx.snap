// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewDetails component renders ViewDetails 1`] = `
<DocumentFragment>
  <div
    class="ViewDetails"
  >
    <h1>
      Bob
    </h1>
    <img
      alt="Bob"
      src="http://img.com/pic.png"
    />
    <div>
      Occupation: builder
    </div>
    <div>
      Weird stuff happened
    </div>
  </div>
</DocumentFragment>
`;
